controllers/selfnoderemediation_controller.go

```diff
@@ -907,3 +912,18 @@ func (r *SelfNodeRemediationReconciler) isResourceDeletionExpired(snr *v1alpha1.
 
 	return true, 0
 }
+
+func (r *SelfNodeRemediationReconciler) getRuntimeStrategy(strategy v1alpha1.RemediationStrategyType) v1alpha1.RemediationStrategyType {
+	if strategy != v1alpha1.AutomaticRemediationStrategy {
+		return strategy
+	}
+
+	if utils.IsOutOfServiceTaintGA {
+		r.logger.Info("Automatically selected OutOfServiceTaint Remediation strategy")
+		return v1alpha1.OutOfServiceTaintRemediationStrategy
+	}
+
+	r.logger.Info("Automatically selected ResourceDeletion Remediation strategy")
```

Comment:
Wouldn't be helpful to state that the selected strategy was selected because the other one is not supported?

Reply:
Basically the same rational as in the prev [comment](https://github.com/medik8s/self-node-remediation/pull/160#discussion_r1391104944)